frontends/web/src/components/language/language.tsx

```diff
@@ -79,11 +80,17 @@ const LanguageSwitch = ({ languages }: TLanguageSwitchProps) => {
 
   const [selectedIndex, setSelectedIndex] = useState<number>(getSelectedIndex(allLanguages, i18n));
   const [activeDialog, setActiveDialog] = useState<boolean>(false);
+  const isMobile = useMatchMedia('(max-width: 768px)');
+
+  const dialogRef = useRef<Dialog>() as React.MutableRefObject<Dialog>;
 
   const changeLanguage = (langCode: TActiveLanguageCodes, index: number) => {
     setSelectedIndex(index);
-    setActiveDialog(false);
     i18n.changeLanguage(langCode);
+    dialogRef.current.deactivate();
```

Comment:
instead of using ref and accessing the deactivate function, could this be passed as prop somehow?

Would something like this could work instead of using ref `<Dialog open={activeDialog}` without the conditional rendering `activeDialog && (` so the component can check props.open and call deactivate, wdyt?

Reply:
Yeah.. this should be possible. I just didn't want to change the usage of all <Dialog/> but it's probably cleaner this way.

I think in fn components, this could very easily be solved using useEffect. We can use `componentDidUpdate` here. Let's see what I can do :)

Reply:
Yeah.. this should be possible. I just didn't want to change the usage of all `<Dialog/>` ,as there's quite a lot of files that are already using Dialog. buuut it's probably cleaner not to do what I did here and alter Dialog a bit, even though that means there's gonna be a lot of small changes in many files.

I think in fn components, this could very easily be solved using useEffect. We can use `componentDidUpdate` here. Let's see what I can do :)